src/tests/spec/corpus.rs

```diff
@@ -59,36 +59,137 @@ fn run_test(test: TestFile) {
     for valid in test.valid {
         let description = format!("{}: {}", test.description, valid.description);
 
-        let bson_to_native_cb = Document::from_reader(
-            &mut hex::decode(&valid.canonical_bson)
-                .expect(&description)
-                .as_slice(),
-        )
-        .expect(&description);
-
-        let mut native_to_bson_bson_to_native_cv = Vec::new();
+        let canonical_bson = hex::decode(&valid.canonical_bson).expect(&description);
```

Comment:
I expanded the corpus tests to cover both deserializers, as they weren't doing so before. This uncovered a few bugs.